model/build-processor/src/main/java/org/keycloak/models/map/processor/GenerateEntityImplementationsProcessor.java

```diff
@@ -275,6 +363,29 @@ public void generate(TypeElement e) throws IOException {
                         }
                     }
                 });
+
+                // Read-only class overrides setters to be no-op
+                pw.println("    public static class Empty " + (an.inherits().isEmpty() ? "" : " extends " + an.inherits()) + " implements " + className + " {");
```

Comment:
Methods won't be undeclared if we implement them in the generator 😆. 

But currently, the Empty class might not be really empty as it will store id for most entities.

Reply:
I am not following. Some of the methods in the entities are not generated and that is the primary reason for having ability to supply parent class in the `GenerateEntityImplementations` annotation, e.g. [MapClientEntity.getClientScopes(boolean)](https://github.com/keycloak/keycloak/blob/38174212f90739e3bc8ec854f28106f1ebdf578c/model/map/src/main/java/org/keycloak/models/map/client/MapClientEntity.java#L58).

Reply:
I understand that, but I believe [these methods]((https://github.com/keycloak/keycloak/blob/38174212f90739e3bc8ec854f28106f1ebdf578c/model/map/src/main/java/org/keycloak/models/map/client/MapClientEntity.java#L41)) should not be part of the `Empty` class as they make it non-empty. For example:
```
MapClientEntity client = MapClientEntityImpl.Empty.INSTANCE;
client.setId("MyId");
client.getId(); // returns "MyId"
```

I would say we should implement all methods in the Empty class including ignored and already implemented in the parent class.

Reply:
I see your point. Still, some of the methods might not be that easy to generate automatically, especially those returning primitive types. I am thinking of unconditional overriding all property accessors (these can be generated automatically), while still allowing for the inheritance that is already there. This would cover the `id` case.

Reply:
A primitive type is used only with `isUpdated`, isn't it? It shouldn't be a problem to treat it in a separate `if` branch.

Reply:
I just realized it is not only the `isUpdated` method, what will we do about this: https://github.com/keycloak/keycloak/blob/main/model/map/src/main/java/org/keycloak/models/map/user/MapUserEntity.java#L238?

Reply:
Keep it implemented from supertype

Reply:
Based on this discussion we had a call and we agreed we will implement all fields accessor methods in the `Empty` class including `isUpdated` so it always returns `false`.